Update Clio install instructions

mDuo13 opened the pull request:
- Clarify nightly and release channels
- Mention self-contained binaries
- Update build from source and default config file links
- Update to reflect that `clio` package now only installs the Clio server; the "all-in-one" package that installs both `rippled` and Clio is no longer supported.
- Fix step numbering and admonition syntax

mDuo13 committed:
Update Clio install instructions

godexsoft commented on docs/infrastructure/installation/install-clio-on-ubuntu.md:
```diff
@@ -10,7 +10,10 @@ labels:
+These instructions install an Ubuntu package that has been compiled and published by Ripple. You can also:
+
+- Download binaries, including for nightly and preview builds, from the [Clio releases page on GitHub](https://github.com/XRPLF/clio/releases/). (Expand the **Assets** section and choose the appropriate version for your OS.)
+- [Build Clio from source](https://github.com/XRPLF/clio/blob/develop/docs/build-clio.md).
```
Maybe worth mentioning that we now also produce docker images: https://hub.docker.com/r/rippleci/clio

godexsoft commented on docs/infrastructure/installation/install-clio-on-ubuntu.md:
```diff
@@ -19,7 +22,9 @@ Before you install Clio, you must meet the following requirements.
+    Clio has the same system requirements as the `rippled` server, except Clio needs less disk space to store the same amount of ledger history.
+    {% /admonition %}
 
 -  You need compatible versions of CMake and Boost. Clio requires C++20 and Boost 1.75.0 or higher.
```
Clio requires C++23 and Boost 1.83.0 or higher. More info here: https://github.com/XRPLF/clio/blob/develop/docs/build-clio.md

godexsoft reviewed (changes requested):
Found a few small things to adjust, looks good otherwise 🙇

mDuo13 committed:
Clio install: update deps & mention Docker images

oeggert commented on docs/infrastructure/installation/install-clio-on-ubuntu.md:
```diff
@@ -19,9 +23,11 @@ Before you install Clio, you must meet the following requirements.
+    {% /admonition %}
 
--  You need compatible versions of CMake and Boost. Clio requires C++20 and Boost 1.75.0 or higher.
+-  You need compatible versions of CMake and Boost. Clio requires C++23 and Boost 1.83.0 or higher.
```
```suggestion
-  Clio requires C++23 and Boost 1.83.0 or higher.
```
Nitpick: just say the requirement.

oeggert commented on docs/infrastructure/installation/install-clio-on-ubuntu.md:
```diff
@@ -130,15 +133,19 @@ Before you install Clio, you must meet the following requirements.
 
-        **Note** You can use multiple `rippled` servers as a data source by add more entries to the `etl_sources` section. If you do, Clio load balances requests across all the servers in the list, and can keep up with the network as long as at least one of the `rippled` servers is synced.
+        {% admonition type="info" name="Note" %}
+        You can use multiple `rippled` servers as a data source by add more entries to the `etl_sources` section. If you do, Clio load balances requests across all the servers in the list, and can keep up with the network as long as at least one of the `rippled` servers is synced.
```
```suggestion
        You can use multiple `rippled` servers as a data source by adding more entries to the `etl_sources` section. If you do, Clio load balances requests across all the servers in the list, and can keep up with the network as long as at least one of the `rippled` servers is synced.
```

oeggert reviewed (approved):
Looks good to me besides a couple minor points.

mDuo13 committed:
Clio install: Apply suggestions from review

Co-authored-by: oeggert <117319296+oeggert@users.noreply.github.com>

mDuo13 committed:
Update Clio install instructions

- Remove metion of build deps from apt install
- Use non-admin WS port in examples
- Add caution about serving public WS
